tests/testthat/test-read_vpts.R

```diff
@@ -1,173 +1,233 @@
+    )
+    # Test if the output is a vpts object
+    expect_true(is.vpts(result))
+  }
+})
+
+test_that("read_vpts() returns error on multiple radars in vp hdf5 files", {
+  skip("Not yet implemented")
+
+  h5_files <- list.files(temp_h5_dir, pattern = "*.h5", full.names = TRUE)
+
   expect_error(
-    read_vpts(vptsfile,
-      wavelength = "a",
-      radar = "KBGM"
-    ),
-    regexp = wavelength_msg,
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      wavelength = -12,
-      radar = "KBGM"
-    ),
-    regexp = wavelength_msg,
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      wavelength = 1:3,
-      radar = "KBGM"
-    ),
-    regexp = wavelength_msg,
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      wavelength = "Q",
-      radar = "KBGM"
-    ),
-    regexp = wavelength_msg,
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      wavelength = "S",
-      radar = "KBGM",
-      sep = "|"
-    ),
-    regexp = "'sep' should be either \",\" or \"\"",
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      sep = NA
-    ),
-    regexp = "'sep' should be either \",\" or \"\"",
-    fixed = TRUE
-  )
-  expect_error(
-    read_vpts(vptsfile,
-      sep = c(",", "&")
-    ),
-    regexp = "'sep' should be either \",\" or \"\"",
-    fixed = TRUE
+    read_vpts(h5_files),
+    "`files` must contain data of a single radar."
   )
+})
+
+test_that("read_vpts() can read remote (gzipped) VPTS CSV files", {
+  skip_if_offline()
+
+  gz_urls <- urls[grepl("\\.gz$", urls)]
+
+  # Test for one file
+  result <- read_vpts(gz_urls[1])
+
+  # Returns vpts class
+  expect_true(is.vpts(result))
+
+  # Test for multiple files
+  result <- read_vpts(gz_urls[1:2])
+
+  # Returns vpts class
+  expect_true(is.vpts(result))
+})
+
+test_that("read_vpts() can read local (gzipped) VPTS CSV files", {
+  skip_if_offline()
+
+  # Test for one file
+  result <- read_vpts(gz_files[1])
+
+  ## Returns vpts class
+  expect_true(is.vpts(result))
+
+  # Test for multiple files
+  result <- read_vpts(gz_files[1:2])
+
+  # Returns vpts class
+  expect_true(is.vpts(result))
+})
+
+test_that("read_vpts() returns error on multiple radars in VPTS CSV files", {
+  skip_if_offline()
+  # Note: this is a limitation until we switch to vpts data frame objects
+
   expect_error(
-    read_vpts(vptsfile,
-      sep = c(",", "||")
-    ),
-    regexp = "'sep' should be either \",\" or \"\"",
-    fixed = TRUE
+    read_vpts(gz_files),
+    "`files` must contain data of a single radar."
   )
 })
 
-test_that("read_vpts() warns for missing wavelength", {
-  expect_warning(
-    read_vpts(vptsfile,
-      radar = "KBGM"
-    ),
-    regexp = "No 'wavelength' argument provided, assuming radar operates at C-band",
-    fixed = TRUE
+# Comapre read_vpts output from data in both formats
+
+if (!require("aws.s3")) {
+  stop("Package 'aws.s3' is not installed. Please install it before proceeding.")
```

Comment:
aws.s3 is in Suggests in DESCRIPTION, so we can expect it to be present on test servers. Instead I would skip these tests when it is not available.